### PR TITLE
CI: Provide prebuilt windows binary for hip

### PR DESCRIPTION
**Pull request by no1wudi**
- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [x] Low
  - [ ] Medium
  - [ ] High

### CODE REVIEW

**Comment by slaren** [comment]
To reduce the time needed for the CI to complete, can you change this so that during pull requests, only one gpu_target is built?

**Comment by no1wudi** [comment]
> To reduce the time needed for the CI to complete, can you change this so that during pull requests, only one gpu_target is built?

OK, how about only build one target for CI and only build all targets on release?

**Comment by slaren** [comment]
Yes that would be good, building all targets for the releases is fine, it's only not necessary for pull requests.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -994,8 +995,74 @@ jobs:
         run: |
           $env:HIP_PATH=$(Resolve-Path 'C:\Program Files\AMD\ROCm\*\bin\clang.exe' | split-path | split-path)
           $env:CMAKE_PREFIX_PATH="${env:HIP_PATH}"
-          cmake -G "Unix Makefiles" -B build -S . -DCMAKE_C_COMPILER="${env:HIP_PATH}\bin\clang.exe" -DCMAKE_CXX_COMPILER="${env:HIP_PATH}\bin\clang++.exe" -DGGML_HIPBLAS=ON
-          cmake --build build --config Release
+          cmake -G "Unix Makefiles" -B build -S . -DCMAKE_C_COMPILER="${env:HIP_PATH}\bin\clang.exe" -DCMAKE_CXX_COMPILER="${env:HIP_PATH}\bin\clang++.exe" -DGGML_HIPBLAS=ON -DCMAKE_BUILD_TYPE=Release -DGGML_RPC=ON
+          cmake --build build -j ${env:NUMBER_OF_PROCESSORS}
+
+  windows-latest-cmake-hip-release:
+    if: ${{ github.event.inputs.create_release == 'true' }}
```
Comment (slaren) [inline .github/workflows/build.yml:1002]: I think this is not going to be enough to create releases automatically on every commit to master, it also needs to check for `github.event_name == 'push' && github.ref == 'refs/heads/master'` like in the other jobs.

**Review by slaren — APPROVED** [review]

**Pull request merged**